link/query.go

```diff
@@ -0,0 +1,63 @@
+package link
+
+import (
+	"fmt"
+	"os"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// QueryOptions defines additional parametes when querying for programs
+// QueryOptions.Path can be a path to a cgroup, netns or LIRC2 device
+type QueryOptions struct {
+	Path        string
+	AttachType  ebpf.AttachType
+	AttachFlags ebpf.AttachFlags
+	QueryFlags  uint32
+}
+
+// QueryPrograms retrieves ProgramIDs associated with the AttachType
```

Comment:
This only works for things that go via RawAttachProgram / BPF_PROG_ATTACH, right? If yes that would be good to document.